source/geometry_model/two_merged_boxes.cc

```diff
@@ -0,0 +1,523 @@
+    }
+
+    template <int dim>
+    bool
+    TwoMergedBoxes<dim>::has_curved_elements() const
+    {
+      return false;
+    }
+
+    template <int dim>
+    void
+    TwoMergedBoxes<dim>::
+    declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Box with lithosphere boundary indicators");
+        {
+          prm.declare_entry ("Lithospheric thickness", "0.2",
+                             Patterns::Double (0),
+                             "The thickness of the lithosphere used to create "
+                             "additional boundary indicators to set specific "
+                             "boundary conditions for the lithosphere. ");
+
+          prm.declare_entry ("X extent", "1",
+                             Patterns::Double (0),
+                             "Extent of the box in x-direction. Units: m.");
+          prm.declare_entry ("Y extent", "1",
+                             Patterns::Double (0),
+                             "Extent of the box in y-direction. Units: m.");
+          prm.declare_entry ("Z extent", "1",
+                             Patterns::Double (0),
+                             "Extent of the box in z-direction. This value is ignored "
+                             "if the simulation is in 2d. Units: m.");
+
+          prm.declare_entry ("Box origin X coordinate", "0",
+                             Patterns::Double (),
+                             "X coordinate of box origin. Units: m.");
+          prm.declare_entry ("Box origin Y coordinate", "0",
+                             Patterns::Double (),
+                             "Y coordinate of box origin. Units: m.");
+          prm.declare_entry ("Box origin Z coordinate", "0",
+                             Patterns::Double (),
+                             "Z coordinate of box origin. This value is ignored "
+                             "if the simulation is in 2d. Units: m.");
+
+          // Lower box repetitions
+          prm.declare_entry ("X repetitions", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in X direction.");
+          prm.declare_entry ("Y repetitions", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in Y direction.");
+          prm.declare_entry ("Z repetitions", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in Z direction.");
+
+          // Upper box repetitions
+          prm.declare_entry ("X repetitions lithosphere", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in X direction in the lithosphere.");
+          prm.declare_entry ("Y repetitions lithosphere", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in Y direction in the lithosphere.");
```

Comment:
I don't think that you can choose the number of X (and in 3d Y) repetitions independently for the two parts of the domain. If you do, you won't be able to match the two parts of the triangulation together any more.